hw/ip/rv_core_ibex/lint/rv_core_ibex.waiver

```diff
@@ -98,14 +94,26 @@ waive -rules CLOCK_USE -location {ibex_id_stage.sv} -regexp {'clk_i' is connecte
+waive -rules {ZERO_REP} -location {ibex_icache.sv} -regexp {Replication count is zero in '{22 - IC_TAG_SIZE{1'b0}}'} \
       -comment "if IC_TAG_SIZE=22, no padding is needed"
-waive -rules {ZERO_REP} -location {ibex_icache.sv} -regexp {Replication count is zero in '{IC_LINE_BEATS_W-1{1'b0}}'} \
+waive -rules {ZERO_REP} -location {ibex_icache.sv} -regexp {Replication count is zero in '{IC_LINE_BEATS_W - 1{1'b0}}'} \
       -comment "if IC_LINE_BEATS=2, IC_LINE_BEATS_W=1 and no padding is needed"
-waive -rules MIXED_SIGN           -location {ibex_lockstep.sv} -regexp {Unsigned operand 'rst_shadow_cnt_q' and signed 'LockstepOffsetW'(1)' encountered in a binary expression} \
+waive -rules {ARITH_CONTEXT} -location {ibex_icache.sv} -regexp {Bitlength of arithmetic operation '(output_addr_q[IC_LINE_W - 1:BUS_W] + {{IC_LINE_BEATS_W - 1{1'b0}},skid_valid_q})' is self-determined in this context} \
+      -comment "The bitlengths are determined but the zero replication count (IC_LINE_BEATS_W=1) seems to confuse the tool"
+waive -rules MIXED_SIGN           -location {ibex_lockstep.sv} -regexp {Unsigned operand 'rst_shadow_cnt_q' and signed '(LockstepOffsetW'(1))' encountered in a binary expression} \
       -comment "In line with our style guide, LockstepOffsetW'(1) results in a signed operand"
 waive -rules UNSIZED_BIT_CONTEXT  -location {ibex_lockstep.sv} -regexp {Bitlength of unsized bit literal "'0" is self determined in this context} \
       -comment {In line with our style guide, writing this as {$bits(delayed_inputs_t){1'b0}} would be much less readable}
+waive -rules CONST_FF             -location {ibex_icache.sv} -regexp {Flip-flop 'reset_inval_q' is driven by constant one} \
+      -comment "The only purpose of the flop is to initiate the cache invalidation after reset"
+waive -rules RESET_DRIVER         -location {ibex_lockstep.sv} -regexp {'(rst_shadow_set_q|rst_shadow_n)' is driven here, and used as an asynchronous reset 'rst_ni' at}
+      -comment "A synchronous counter is needed to release the shadow core reset with a delay of LockstepOffset clock cycles"
+waive -rules RESET_DRIVER         -location {ibex_lockstep.sv} -regexp {'(rst_shadow_set_q|rst_shadow_n)' driven in module 'ibex_lockstep'}
+      -comment "A synchronous counter is needed to release the shadow core reset with a delay of LockstepOffset clock cycles"
+waive -rules RESET_MUX            -location {ibex_lockstep.sv} -regexp {Asynchronous reset 'rst_shadow_n' is driven by a multiplexer here}
+      -comment "The test enable input used to control the bypass can be considered static"
+waive -rules RESET_USE            -location {ibex_lockstep.sv} -regexp {'rst_shadow_set_q' is used for some other purpose, and as asynchronous reset 'rst_ni' at}
+      -comment "A synchronous counter is needed to release the shadow core reset with a delay of LockstepOffset clock cycles and start the comparison logic one clock cycle later"
```

Comment:
@GregAC @msfschaffner I've reviewed the mechanism for generating the reset of the lockstep core and it looks okay to me. Our implementation follows common guidelines. I think it's safe to waive these lint errors.

Reply:
The mechanisms look fine to me, the only question I would have is whether we should use `prim_clock_mux2` and `prim_flop` to implement it.
Specifically, for other scan bypasses, we are using primitives as well, e.g.:
https://github.com/lowRISC/opentitan/blob/d97b4a0fe8c4370169799cb4b2783833b716e26f/hw/ip/lc_ctrl/rtl/lc_ctrl.sv#L170-L188

Reply:
This makes sense Michael, I've filed an Ibex PR to achieve this. See lowRISC/Ibex#1505, PTAL.